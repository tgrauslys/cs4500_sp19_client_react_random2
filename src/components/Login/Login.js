import React from 'react'

const Login = ({login, updateUsername, updatePassword, isErrorMessageOn}) => {
    let errorMessage = <div className="col-12">Invalid login information</div>
    if (!isErrorMessageOn) {
        errorMessage = ''
    }

    return (<div className="container">
        <h1>Welcome back</h1>
        <br/>
        <div>
            <div className="row">
                <div className="col-12">
                    <label for="email">Email</label>
                    <input id="email" className="form-control" onChange={e => updateUsername(e)} />
                </div>
                <div className="col-12">
                    <br/>
                    <label for="password">Password</label>
                    <input id="password" className="form-control"onChange={e => updatePassword(e)}/>
                </div>
            </div>
            <br/>
            <div className="row">
                <div className="col-6">
                    <label for="password">
                        <input type="checkbox"/>
                        &nbsp; Remember me
                    </label>
                </div>
                <div className="col-6">
                    <a className="float-right" href="#">Forgot password?</a>
                </div>
            </div>
            <br/>
            <div className="row">
                <div className="col-12">
                    {errorMessage}
                </div>
            </div>
            <div className="row">
                <div className="col-12">
                    <button onClick={login}
                        style={{margin: '2px'}}
                        type="button" className="btn btn-primary btn-block">Log in</button>
                </div>
            </div>
        </div>
    </div>)
}
    

export default Login