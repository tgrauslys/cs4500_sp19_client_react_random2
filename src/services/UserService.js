export default class UserService {
    static instance = null;
    static getInstance() {
        if(UserService.instance === null) {
            UserService.instance = new UserService()
        }
        return this.instance
    }
    findUserById = userId =>
        fetch(`http://localhost:8080/api/users/${userId}`)
            .then(response => response.json())
    findAllUsers = () =>
        fetch("http://localhost:8080/api/users")
            .then(response => response.json())
}
