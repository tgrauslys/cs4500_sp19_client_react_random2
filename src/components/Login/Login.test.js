import registry from '../../data/Register.mock.json'
import React from 'react'
import Login from './Login'
import TestRenderer from 'react-test-renderer'
import UserService from '../../services/UserService'

const userService = UserService.getInstance()

describe('login page', () => {
    test('render login screen', async () => {
        let updateUsername = () => {}
        let updatePassword = () => {}
        let submit = () => {}

        const testRenderer = TestRenderer.create(
            <Login
                login={submit}
                updateUsername={updateUsername}
                updatePassword={updatePassword}
                isErrorMessageOn={true}/>
        )

        let tree = testRenderer.toJSON()
        expect(tree).toMatchSnapshot()

        const testInstance = testRenderer.root

        const usernameField = testInstance.findByProps({id: "username"})
        const passwordField = testInstance.findByProps({id: "password"})
        const checkbox = testInstance.findByProps({type: "checkbox"})
        const forgotPasswordLink = testInstance.findByProps({href: "#"})
        const loginButton = testInstance.findByProps({className: "btn btn-primary btn-block"})
        const fields = testInstance.findAllByProps({className: "form-control"})

        expect(usernameField).toBeDefined()
        expect(passwordField).toBeDefined()
        expect(checkbox).toBeDefined()
        expect(forgotPasswordLink).toBeDefined()
        expect(loginButton).toBeDefined()
        expect(loginButton.type).toBe("button")
        expect(fields.length).toBe(2)
    })
})