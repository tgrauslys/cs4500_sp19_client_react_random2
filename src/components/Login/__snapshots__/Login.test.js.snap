// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page render login screen 1`] = `
<div
  className="container"
>
  <h1>
    Welcome back
  </h1>
  <br />
  <div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <label
          for="username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="username"
          onChange={[Function]}
        />
      </div>
      <div
        className="col-12"
      >
        <br />
        <label
          for="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
        />
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <label
          for="password"
        >
          <input
            type="checkbox"
          />
          Â  Remember me
        </label>
      </div>
      <div
        className="col-6"
      >
        <a
          className="float-right"
          href="#"
        >
          Forgot password?
        </a>
      </div>
    </div>
    <br />
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        Invalid login information
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <button
          className="btn btn-primary btn-block"
          onClick={[Function]}
          style={
            Object {
              "margin": "2px",
            }
          }
          type="button"
        >
          Log in
        </button>
      </div>
    </div>
  </div>
</div>
`;
